### ISSUE TITLE
DevMojoIT hangs if the test build process fails

### ISSUE THREAD

**Issue by starksm64** [quarkusio/quarkus#1773]
One of my PRs kept hanging and timing out during the io.quarkus.maven.it.DevMojoIT tests. In tracking down the issue, I found that if the target test project that the DevMojoIT is building fails to compile, the DevMojoIT test fails on a timeout. DevMojoIT is not detecting the failed build. The reason the build was failing is that a dependency on the quarkus-maven-plugin was added to the core test-extension, and this brought the build of the devtools module to before any of the extension modules, so none of the extension dependencies existed on a system with no previous io.quarkus artifacts.

The steps to cause the DevMojoIT to hang are:
1. Do a clean build of the quarkus repository
2. Go into the devtools/maven module
3. Remove the quarkus-resteasy;  ~/.m2/repository/io/quarkus/quarkus-resteasy
4. Verify that the target/test-classes/projects/project-classic-run will now fail to compile;
cd target/test-classes/projects/project-classic-run
mvn compile
5. Run the testThatClassAppCanRun test:
Scotts-iMacPro:maven starksm$ mvn -Dit.test=io.quarkus.maven.it.DevMojoIT#testThatClassAppCanRun verify

This will timeout.

This could be a duplicate of #1219, but in this case I know how to reproduce the issue.

**Comment by cescoffier** [comment]
This is very annoying. Even catching the error won't make it better (won't timeout but will fail).
We can add an artificial dependency in the maven pom, just to make ordering OK. But, I'm not a big fan of that. Unfortunately, the other solution would be do drop the ITs which are our only way to verify things work.

**Comment by cescoffier** [comment]
@starksm64 I've built master from an empty maven repo and didn't get the issue.

```
...
[INFO] Quarkus - Core - Test Extension .................... SUCCESS [  0.017 s]
[INFO] Quarkus - Core - Test Extension - Runtime .......... SUCCESS [  0.531 s]
[INFO] Quarkus - Core - Test Extension - Deployment ....... SUCCESS [  4.969 s]
...
[INFO] Quarkus - Maven Plugin ............................. SUCCESS [05:29 min]
```

Do you know where the introduction of the dependency (Core Test -> Quarkus Maven plugin) has been introduced?

**Comment by starksm64** [comment]
Yes, it was in the core/test-extension when I added a native image test. I have moved that out into integration-tests so it has been resolved.

Shouldn't we be catching the error and failing the build though? It was a bug in the test-extension to add a dependency on the quarkus-maven-plugin given that it tests projects that depend on extensions.

We could move the devtools tests that depend on extensions into the integration-tests to ensure that a dependency on the quarkus-maven-plugin does not cause tests that may depend on extensions that have not been built yet to be run.
